### ISSUE TITLE
adding hotkeys

### ISSUE THREAD

**Issue by han-so1omon** [drobilla/ingen#6]
i want to use ingen for live mixing. is there a way to add hotkeys? or maybe i can add it in with a pr if you give me a boilerplate method to do so

**Comment by drobilla** [comment]
Hotkeys for what sort of thing?

**Comment by han-so1omon** [comment]
For navigating between nodes and adjusting parameters it triggering actions quickly. I want to use ingen for live looping with sooperlooper and such

**Comment by drobilla** [comment]
That's probably going to be quite some work (and mostly in ganv, which I hope to replace anyway), but it would be doable.  I'd certainly like to see it, but navigation a graph by keyboard seems pretty tricky, even conceptually.  Do you have any ideas of how you'd like this to work in mind (spatial, mnemonic, ...)?

**Comment by han-so1omon** [comment]
I'm doing it in ingen gui. I'm new to gui programming, but I'm down to give it a try. Most of the code can work in any program with the same basic node representation and GTK tools.

Here's my plan (vim-like):

(1) assign a number to each node
(2) another window popup that displays node numbers
(3) revise each node to display a number before each option (in,out, volume, etc)
(4) create simple key combination buffer to detect next command. clear buffer on command key or reserved key {n,o, h, j,shift}

Key combinations:
(1) key combination <num> + n <- go to node <num>
(2) key combination <num> + o <- go to option <num>
(3) key <j,h> to slowly increase/decrease current option
(4) key shift + <j,h> to quickly increase/decrease current option

**Comment by drobilla** [comment]
I see.  I would avoid popping up a window to display node numbers though, it would be really tedious to correlate with the node you're looking at and want to control.  I think it would be better to draw labels on the nodes themselves.  Maybe when some modifier is pressed, like the various keyboard navigation plugins for browsers.

Not sure about numbers vs just using the existing human names somehow, but I guess it would be faster.

**Comment by han-so1omon** [comment]
Yeah but not all nodes are visible at all times. Maybe a modifier to show all nodes with names
